client/state/data-layer/wpcom-http/actions.js

```diff
@@ -17,28 +82,57 @@ import { WPCOM_HTTP_REQUEST } from 'state/action-types';
  * @param {Object} [onProgress] Redux action to call on progress events from an upload
  * @param {Object} [options] extra options to send to the middleware, e.g. retry policy or offline policy
  * @param {Object} [action] default action to call on HTTP events
- * @returns {Object} Redux action describing WordPress.com API HTTP request
+ * @return {Object} Redux action describing WordPress.com API HTTP request
  */
-export const http = ( {
-	apiVersion = 'v1',
-	body = {},
-	method,
-	path,
-	query = {},
-	formData,
-	onSuccess,
-	onFailure,
-	onProgress,
-	...options,
-}, action = null ) => ( {
-	type: WPCOM_HTTP_REQUEST,
-	body,
-	method,
-	path,
-	query: { ...query, apiVersion },
-	formData,
-	onSuccess: onSuccess || action,
-	onFailure: onFailure || action,
-	onProgress: onProgress || action,
-	options,
-} );
+export const http = ( request, action = null ) => {
+	const {
+		apiVersion,
+		apiNamespace,
+		body = {},
+		method,
+		formData,
+		path,
+		query = {},
+		onSuccess,
+		onFailure,
+		onProgress,
+		...options
+	} = request;
+
+	// validate request and throw on error
+	validators.forEach( validator => validator( request ) );
+
+	const actionProperties = {
+		body,
+		onSuccess: onSuccess || action,
+		onFailure: onFailure || action,
+		onProgress: onProgress || action,
+		query: { ...query },
+		options,
+	};
```

Comment:
this object seems unnecessary - why not simply put these properties directly in the return statement below?